app.go

```diff
@@ -0,0 +1,423 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"net/http"
+	"time"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+type AsyncTasksApp struct {
+	db     *DBConnection
+	router *mux.Router
+}
+
+func NewAsyncTasksApp(db *DBConnection, router *mux.Router) *AsyncTasksApp {
+	app := &AsyncTasksApp{
+		db:     db,
+		router: router,
+	}
+
+	app.InitRoutes()
+
+	return app
+}
+
+func (a *AsyncTasksApp) InitRoutes() {
```

Comment:
It may be wise, eventually, to do this whole section more programmatically, based on a data structure describing the routes. But for now it's fairly small.